src/workerd/api/trace.h

```diff
@@ -190,6 +191,24 @@ class TraceItem::AlarmEventInfo final: public jsg::Object {
   const Trace::AlarmEventInfo& eventInfo;
 };
 
+class TraceItem::QueueEventInfo final: public jsg::Object {
+public:
+  explicit QueueEventInfo(kj::Own<Trace> trace, const Trace::QueueEventInfo& eventInfo);
+
+  kj::StringPtr getQueueName();
+  uint32_t getBatchSize();
+  // TODO(now): Add something about the timestamp(s) of the newest/oldest message(s) in the batch?
```

Comment:
I guess timestamps are planned as a next step?  I think for other event types, we're just packing in enough conveniently available information into the eventinfo structures to disambiguate the requests.  Although I suppose users always have the fallback option of `console.log()`ing info specific to their application.

Reply:
Thanks for catching that I had missed this!

It isn't totally obvious what the best way to disambiguate the invocations is, since each likely contains a whole bunch of messages. So one thought here was to report info about the oldest/newest timestamps in the batch, or to report the oldest/newest message IDs or something else along those lines. But I had meant to leave that out for now and just potentially follow up with something later.

I'll push the removal of this after CI finishes running since I don't want to interrupt it after it's been running for 90+ minutes just to remove a comment.

Reply:
Man this CI is slow as molasses... 3.5 hours after starting and it's still going 🐌